test_vm/src/lib.rs

```diff
@@ -64,26 +74,134 @@ impl<'bs> VM<'bs> {
             actors_dirty: RefCell::new(false),
             actors_cache: RefCell::new(HashMap::new()),
             empty_obj_cid: empty,
+            network_version: NetworkVersion::V16,
+            curr_epoch: ChainEpoch::zero(),
+            invocations: RefCell::new(vec![]),
         }
     }
 
     pub fn new_with_singletons(store: &'bs MemoryBlockstore) -> VM<'bs> {
         // craft init state directly
         let v = VM::new(store);
+
+        // system
+        let sys_st = SystemState::new(store).unwrap();
+        let sys_head = v.put_store(&sys_st);
+        v.set_actor(
+            *SYSTEM_ACTOR_ADDR,
+            actor(*SYSTEM_ACTOR_CODE_ID, sys_head, 0, TokenAmount::from(1_000_000u32)),
```

Comment:
Why give the system actor tokens?

Reply:
Or, well, why give any of these actors tokens...

Reply:
Tests need a faucet and its nice to minimize manual actor setting.  To make this better I moved to an explicit faucet account actor that consumes all of the system funds by the end of the call.

I also set up the reward singleton with funding matching current protocol parameters so that it behaves as expected.